### PR TITLE
Check for events that are open for registration

### PR DESCRIPTION
**Pull request by dominikglandorf**
Needs the corresponding backend pull request.

Will check for the isOpenForRegistration flag for events to register as participant. This is for opening the registration.

### CODE REVIEW

File: src/client/react/App.jsx [hunk 1]
```diff
@@ -68,6 +71,25 @@ class App extends React.Component {
...
+          this.setState({
+            isRequestingOpenRegistration: false,
+            isRegistrationOpen: false
+          });
+        }
+      })
+      .catch(err => {
+        console.error('Error getting events', err);
+      });
+  }
```
Comment (florianschmidt1994) [inline src/client/react/App.jsx]: Maybe use async function and await with try / catch?

File: src/client/react/App.jsx [hunk 2]
```diff
@@ -199,10 +225,6 @@ class App extends React.Component {
   }
 }
 
-function registrationIsLocked() {
-  return Date.now() > new Date('Tue May 09 2017 05:00:00 GMT+0200 (CEST)');
-}
-
```
Comment (florianschmidt1994) [inline src/client/react/App.jsx:205]: Nice!

**Comment by dominikglandorf** [comment]
https://github.com/BreakOutEvent/breakout-backend/pull/203 is neccessary.

File: src/server/controller/RegistrationController.js [hunk 3]
```diff
@@ -52,14 +52,17 @@ const sendErr = (res, errMsg, err) => {
...
+  const events = yield api.getAllEvents();
+  if (events.find(event => event.openForRegistration)) {
     next();
+  } else {
+    res.redirect('/closed');
   }
-};
+}).catch(err => {
+  sendErr(res, err.message, err);
+});
```
Comment (florianschmidt1994) [inline src/server/controller/RegistrationController.js]: Don't use a co function here but a generator function `registration.lock = function*() { ... }` The router will wrap this automatically

**Comment by florianschmidt1994** [comment]
Looks pretty good! Address the minor comments and this can be merged

**Pull request merged**